packages/teams-js/src/internal/telemetry.ts

```diff
@@ -50,6 +56,60 @@ export const enum ApiVersionNumber {
   V_3 = 'v3',
 }
 
+/**
+ * @internal
+ * Limited to Microsoft-internal use
+ */
+export class HostToAppMessageDelayTelemetry {
+  private static callbackInformation: Map<MessageUUID, CallbackInformation> = new Map();
+
+  /**
+   * @internal
+   * Limited to Microsoft-internal use
+   *
+   * Store information about a particular message.
+   * @param messageUUID The message id for the request.
+   * @param callbackInformation The information of the callback.
+   */
+  public static storeCallbackInformation(messageUUID: MessageUUID, callbackInformation: CallbackInformation): void {
+    HostToAppMessageDelayTelemetry.callbackInformation.set(messageUUID, callbackInformation);
+  }
+
+  /**
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  public static clearMessages(): void {
+    HostToAppMessageDelayTelemetry.callbackInformation.clear();
+  }
+
+  /**
+   * @internal
+   * Limited to Microsoft-internal use
+   *
+   * Executes telemetry actions related to host to app performance metrics.
+   * @param callbackId The message id for the request.
+   * @param message The response from the host.
+   * @param logger A logger for logging any possible error.
+   */
+  public static telemetryHostToAppPerformanceMetrics(
+    callbackID: MessageUUID,
+    message: MessageResponse,
+    logger: debug.Debugger,
+  ): void {
+    const callbackInformation = HostToAppMessageDelayTelemetry.callbackInformation.get(callbackID);
+    if (callbackInformation && message.timestamp) {
+      handleHostToAppPerformanceMetrics({
+        actionName: callbackInformation.name,
+        messageDelay: getCurrentTimestamp() - message.timestamp,
```

Comment:
from you local testing, can you please add some data points for values recorded both for two-way and one-way messages